lib/Word_Lib/More_Word.thy

```diff
@@ -216,16 +213,12 @@ lemma of_int_uint:
 corollary word_plus_and_or_coroll:
   "x AND y = 0 \<Longrightarrow> x + y = x OR y"
   for x y :: \<open>'a::len word\<close>
-  using word_plus_and_or[where x=x and y=y]
-  by simp
+  by (fact disjunctive_add2)
 
 corollary word_plus_and_or_coroll2:
   "(x AND w) + (x AND NOT w) = x"
   for x w :: \<open>'a::len word\<close>
-  apply (subst disjunctive_add)
-   apply (simp add: bit_simps)
-  apply (simp flip: bit.conj_disj_distrib)
-  done
+  by (fact and_plus_not_and)
```

Comment:
silly question: what does the `fact` tactic do?

Reply:
My memory is hazy, but I think it's pretty much equivalent to `rule`, but does a more elaborate lookup of facts from the local proof context. I.e. you can say `by fact` and get things you have proved with `have` without name etc.